python_lib/qss_compiler/examples/TestOps.py

```diff
@@ -0,0 +1,144 @@
+#
+#     http:#www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+# Example python script for generating mlir for a variety of Pulse Ops
+
+from qss_compiler.mlir.ir import InsertionPoint, Location, Module, Context
+from qss_compiler.mlir.ir import IntegerType, F64Type, ComplexType
+from qss_compiler.mlir.ir import DenseElementsAttr
+
+from qss_compiler.mlir.dialects import arith, complex, func
+from qss_compiler.mlir.dialects import pulse, quir
+
+import numpy as np
+
+with Context(), Location.unknown():
+    # register dialects
+    pulse.pulse.register_dialect()
+    quir.quir.register_dialect()
+
+    # declare types
+    i1 = IntegerType.get_signless(1)
+    i32 = IntegerType.get_signless(32)
+    f64 = F64Type.get()
+
+    c64 = ComplexType.get(f64)
+
+    p = pulse.PortType.get()
+    wf = pulse.WaveformType.get()
+    mf = pulse.MixedFrameType.get()
+    f = pulse.FrameType.get()
+
+    # create module
+    module = Module.create()
+    with InsertionPoint(module.body):
+        # create main func op
+        function = func.FuncOp("main", ([], [i1, i1]))
+        function.add_entry_block()
+
+        # create test sequence op
+        seq = pulse.SequenceOp("test_pulse_ops", [p, p, f, mf, c64], [i1, i1])
+        seq.add_entry_block()
+    # define sequence
+    with InsertionPoint(seq.entry_block):
+        # define args
+        p0, p1, f0, mf0, amp = seq.entry_block.arguments
+
+        # # define constants
+        dur = arith.ConstantOp(i32, 160)
+        sigma = arith.ConstantOp(i32, 40)
+        width = arith.ConstantOp(i32, 1000)
+        beta = arith.ConstantOp(f64, -1.3677586253287046)
+
+        samples = np.array([[0.0, 0.5], [0.5, 0.5], [0.5, 0.0]])
+        samples_2d = DenseElementsAttr.get(samples)
+
+        # # create waveforms
+        gauss = pulse.GaussianOp(dur, amp, sigma)
+        gauss_square = pulse.GaussianSquareOp(dur, amp, sigma, width)
+        drag = pulse.DragOp(dur, amp, sigma, beta)
+        const = pulse.ConstOp(dur, amp)
+        kernel_waveform = pulse.Waveform_CreateOp(samples_2d)
+
+        # mixed frame
+        mf1 = pulse.MixFrameOp(p0, "mf1-p0")
+
+        # define complex amp
+        param_amp_r = arith.ConstantOp(f64, 0.10086211860780928)
+        param_amp_i = arith.ConstantOp(f64, 0.0012978777572167797)
+        param_amp = complex.CreateOp(ComplexType.get(F64Type.get()), param_amp_r, param_amp_i)
+
+        # freq stuff
+        # define freq
+        fc = arith.ConstantOp(f64, 200.0e4)
+
+        pulse.SetFrequencyOp(mf0, fc)
+        pulse.SetFrequencyOp(f0, fc)
+
+        pulse.ShiftFrequencyOp(mf0, fc)
+        pulse.ShiftFrequencyOp(f0, fc)
+
+        # phase stuff
```

Comment:
```suggestion
        # phase operations
```

Reply:
Updated in https://github.com/openqasm/qe-compiler/pull/276/commits/9393557041fc7ef5c563c97aaaab4f734a4c9b55